pysus/online_data/SIA.py

```diff
@@ -133,18 +136,45 @@ def _fetch_file(fname, ftp, ftype):
     :param ftype: file type: DBF|DBC
     :return: pandas dataframe
     """
-    print(f"Downloading {fname}...")
+
+    multiples = False
     try:
-        ftp.retrbinary(f"RETR {fname}", open(fname, "wb").write)
-    except:
-        try:
-            ftp.retrbinary(f"RETR {fname.lower()}", open(fname, "wb").write)
-        except:
-            raise Exception(f"File {fname} not available")
-    if ftype == "DBC":
-        df = read_dbc(fname, encoding="iso-8859-1")
-    elif ftype == "DBF":
-        dbf = DBF(fname, encoding="iso-8859-1")
-        df = pd.DataFrame(list(dbf))
+        fnames = check_file_split(fname, ftp)
+        if len(fnames) > 1:
+            multiples = True
+        for fn in fnames:
+            fnfull = os.path.join(CACHEPATH, fn)
+            print(f"Downloading {fn}...")
+            fobj = open(fnfull, "wb")
+            ftp.retrbinary(f"RETR {fn}", fobj.write)
+            dbc2dbf(fnfull, fnfull.replace('.dbc', '.dbf'))
+            os.unlink(fnfull)
+    except Exception as exc:
+        raise Exception(f"Retrieval of file {fn} failed with the following error:\n {exc}")
+    if multiples:
+        print(f"This download is split into the following files: {fnames}\n"
+              f"They have been downloaded in {CACHEPATH}.\n"
+              f"To load them, use the pysus.utilities.read_dbc_dbf function.")
+        return
+    df = read_dbc_dbf(fname)
+
     os.unlink(fname)
     return df
```

Comment:
Considering that it may be necessary to discover and process files in the multi-file scenario, wouldn't it be better to split the download function in two and make the code easier in both scenarios?

Trying to maintain code compatibility, my suggestion would be:

```
def download_sia_dbfs(<same function download signature>):
  <same code between line 43 and 153>
  # return a list of dbf file names
  if multiples:
    return fnames
  return [fname,] 

```

The download function could be:

```
def download(<same signature>):
  
  # return a list of dbf file names
  fnames = download_sia_dbfs(<same signature>)
  if len(fnames) > 1:
        print(f"This download is split into the following files: {fnames}\n"
              f"They have been downloaded in {CACHEPATH}.\n"
              f"To load them, use the pysus.utilities.read_dbc_dbf function.")
        return
    fname = fnames[0]
    df = read_dbc_dbf(fname)
    os.unlink(fname)
    return df
```

Reply:
thanks for the suggestion. Done.